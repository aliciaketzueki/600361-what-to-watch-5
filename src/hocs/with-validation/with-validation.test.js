import React, {Fragment} from "react";
import renderer from "react-test-renderer";
import withValidation from "./with-validation";
import PropTypes from "prop-types";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withValidation(MockComponent);

describe(`Render withValidation`, () => {
  it(`Should withValidation render correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped>
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
