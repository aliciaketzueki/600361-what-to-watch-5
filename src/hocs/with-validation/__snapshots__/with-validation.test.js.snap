// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withValidation Should withValidation render correctly 1`] = `<div />`;
